plasma_framework/contracts/src/exits/utils/BondSize.sol

```diff
@@ -26,52 +26,57 @@ library BondSize {
         uint16 upperBoundMultiplier;
     }
 
-    function buildParams(uint128 _initialBondSize, uint16 _lowerBoundDivisor, uint16 _upperBoundMultiplier)
+    function buildParams(uint128 initialBondSize, uint16 lowerBoundDivisor, uint16 upperBoundMultiplier)
         internal
         pure
         returns (Params memory)
     {
         // Set the initial value to far in the future
         uint128 initialEffectiveUpdateTime = 2 ** 63;
+
+        require(initialBondSize != 0, "initialBondSize cannot be 0");
+        require(lowerBoundDivisor != 0, "lowerBoundDivisor cannot be 0");
+        require(upperBoundMultiplier != 0, "upperBoundMultiplier cannot be 0");
+
         return Params({
-            previousBondSize: _initialBondSize,
+            previousBondSize: initialBondSize,
             updatedBondSize: 0,
             effectiveUpdateTime: initialEffectiveUpdateTime,
-            lowerBoundDivisor: _lowerBoundDivisor,
-            upperBoundMultiplier: _upperBoundMultiplier
+            lowerBoundDivisor: lowerBoundDivisor,
+            upperBoundMultiplier: upperBoundMultiplier
         });
     }
 
     /**
     * @notice Updates the bond size.
-    * @notice The new value is bounded by 0.5 and 2x of current bond size.
-    * @notice There is a waiting period of 2 days before the new value goes into effect.
+    * @dev There is a waiting period of 2 days before the new value goes into effect.
     * @param newBondSize the new bond size.
     */
-    function updateBondSize(Params storage _self, uint128 newBondSize) internal {
-        validateBondSize(_self, newBondSize);
+    function updateBondSize(Params storage self, uint128 newBondSize) internal {
+        validateBondSize(self, newBondSize);
 
-        if (_self.updatedBondSize != 0 && now > _self.effectiveUpdateTime) {
-            _self.previousBondSize = _self.updatedBondSize;
+        if (self.updatedBondSize != 0 && now >= self.effectiveUpdateTime) {
```

Comment:
This `now >= self.effectiveUpdateTime` part was a bug! It was `now > self.effectiveUpdateTime` but in `bondSize(self)` it is using `now < self.effectiveUpdateTime`. As a result, when `now == self.effectiveUpdateTime` it would fail to fall into this `if` block.